transformer_engine/pytorch/csrc/extensions.h

```diff
@@ -605,3 +605,45 @@ size_t get_cudnn_version();
 bool userbuf_comm_available();
 
 void placeholder();
+
+
+/***************************************************************************************************
+ * Support THD format for Context Parallel
+ **************************************************************************************************/
+
+at::Tensor thd_read_half_tensor(const at::Tensor &tensor,
+                                const at::Tensor &cu_seqlens,
+                                int half_idx
+);
+
+void thd_lse_correction(at::Tensor &lse,
+                        const at::Tensor &lse_per_step,
+                        const at::Tensor &cu_seqlens,
+                        int total_tokens
+);
+
+at::Tensor thd_read_half_lse(const at::Tensor &lse,
+                             const at::Tensor &cu_seqlens,
+                             int total_tokens
+);
+
+void thd_out_correction(at::Tensor &out,
+                        const at::Tensor &out_per_step,
+                        const at::Tensor &lse,
+                        const at::Tensor &lse_per_step,
+                        const at::Tensor &cu_seqlens,
+                        bool is_half
```

Comment:
I think `is_half` here means second half of output tensor? basically I mean the argument naming does not represent the exact functionality?

Reply:
Changed the name to `bool only_second_half`